cli/command/service/logs.go

```diff
@@ -212,43 +235,90 @@ type logWriter struct {
 	opts *logsOptions
 	f    *taskFormatter
 	w    io.Writer
+
+	// detailsCache saves a pre-sorted list of the details keys, so we don't
+	// have to recompute their order every time. this operates under the
+	// assumption that the details keys are always the same (they should be)
```

Comment:
I actually think we can, for now? Only way I can see this not working is if the log stream contained many different services together, which would have different log options.